### ISSUE TITLE
More efficient noise implementation

### ISSUE THREAD

**Issue by JoeCreates** [johndimi/djFlixel#11]
Instead of copyPixels every frame and self generating the noise, you can make use of extsing functinos in flixel (for animation) and openfl (for the noise itself)

```
	noiseSprite = new FlxSprite();
		var frames = 5;
		var noiseFrames:BitmapData = new BitmapData(Std.int(width * frames / 2), Std.int(height / 2), false);
		noiseFrames.noise(FlxG.random.int(), 20, 190);
		noiseSprite.loadGraphic(noiseFrames, true, Std.int(width / 2), Std.int(height / 2), false, "cloneCamNoise");
		noiseSprite.animation.add("noise", [for (i in 0...frames) i], 20);
		noiseSprite.animation.play("noise");
```

**Comment by johndimi** [comment]
Hey! It's been a long time since I touched djFlixel.

`copypixels()` is already low level enough, and flixel itself uses it to update animations. `FlxFrame.paint()` is the function the puts new framedata to a sprite and it uses copypixels. I think that FlxAnimation is just extra overhead that I don't need for such a simple 1:1 buffer copy.

About using the built in noise generator of `bitmapdata` , I think I chose this simpler route so I could generate and color the pixels in one go and have them play raw without needing any colorTransform or filters later in order to achieve the desired noise colors *(I only needed 2 colors)*. Also using a custom thing I have more control like the probability ratio, I don't think I can achieve that with the `bitmapdata.noise()` ? 🤔 

I am eventually going to get back and start working on this lib again. 
Perhaps I'll look into noisegen again then 😄

Thanks!

**Comment by JoeCreates** [comment]
Copypixels is only used for flash and is slow on native. I believe you can
do only 2 colors easily with bitmapdata noise but I'll have to check
tomorrow. :)

**Comment by johndimi** [comment]
Yes, I could be wrong. I haven't tested on native builds at all. 😟 I used to test on native flash and some html5, (windows builds through adobe air) So I don't know how fast/slow things could be there. I always figured that native C++ would always be faster at anything compared to html5/flash ! 

It seems that `bitmapdata.draw` can utilize the GPU, where `bitmapdata.copypixels` is being done by the CPU ? I guess the safest solution is what you suggested and use an `FlxAnimation` to handle everything! 

I'll check this again, and experiment with the built in noise generator, see how it goes 👍
